designdocs/capacity.md

```diff
@@ -23,89 +23,105 @@ Frequency explains how Capacity can be acquired through staking, refills, and us
+   + Mutate<Self::AccountId>
+   + InspectFreeze<Self::AccountId>
+   + InspectFungible<Self::AccountId>;
 
-    /// Function that checks if an MSA is a valid target.
-    type TargetValidator: TargetValidator;
+  /// Function that checks if an MSA is a valid target.
+  type TargetValidator: TargetValidator;
 
-    /// The minimum required token amount to stake. It facilitates cleaning dust when unstaking.
-    #[pallet::constant]
-    type MinimumStakingAmount: Get<BalanceOf<Self>>;
+  /// The minimum required token amount to stake. It facilitates cleaning dust when unstaking.
+  #[pallet::constant]
+  type MinimumStakingAmount: Get<BalanceOf<Self>>;
 
-    /// The maximum number of unlocking chunks a StakingAccountLedger can have.
-    /// It determines how many concurrent unstaked chunks may exist.
-    #[pallet::constant]
-    type MaxUnlockingChunks: Get<u32>;
+  /// The minimum required token amount to remain in the account after staking.
+  #[pallet::constant]
+  type MinimumTokenBalance: Get<BalanceOf<Self>>;
 
-    #[cfg(feature = "runtime-benchmarks")]
-    /// A set of helper functions for benchmarking.
-    type BenchmarkHelper: RegisterProviderBenchmarkHelper;
+  /// The maximum number of unlocking chunks a StakingAccountLedger can have.
+  /// It determines how many concurrent unstaked chunks may exist.
+  #[pallet::constant]
+  type MaxUnlockingChunks: Get<u32>;
 
-    /// The number of Epochs before you can unlock tokens after unstaking.
-    #[pallet::constant]
-    type UnstakingThawPeriod: Get<u16>;
+  #[cfg(feature = "runtime-benchmarks")]
+  /// A set of helper functions for benchmarking.
+  type BenchmarkHelper: RegisterProviderBenchmarkHelper;
 
-    /// Maximum number of blocks an epoch can be
-    /// currently used as the actual value of epoch length.
-    #[pallet::constant]
-    type MaxEpochLength: Get<BlockNumberFor::<Self>>;
+  /// The number of Epochs before you can unlock tokens after unstaking.
+  #[pallet::constant]
+  type UnstakingThawPeriod: Get<u16>;
 
-    /// A type that provides an Epoch number
-    /// traits pulled from frame_system::Config::BlockNumber
-    type EpochNumber: Parameter
-        + Member
-        + MaybeSerializeDeserialize
-        + MaybeDisplay
-        + AtLeast32BitUnsigned
-        + Default
-        + Copy
-        + sp_std::hash::Hash
-        + MaxEncodedLen
-        + TypeInfo;
-}trait Config
+  /// Maximum number of blocks an epoch can be
+  #[pallet::constant]
+  type MaxEpochLength: Get<BlockNumberFor<Self>>;
+
+  /// A type that provides an Epoch number
+  /// traits pulled from frame_system::Config::BlockNumber
+  type EpochNumber: Parameter
+   + Member
+   + MaybeSerializeDeserialize
+   + MaybeDisplay
+   + AtLeast32BitUnsigned
+   + Default
+   + Copy
+   + sp_std::hash::Hash
+   + MaxEncodedLen
+   + TypeInfo;
+
+  /// How much FRQCY one unit of Capacity costs
+  #[pallet::constant]
+  type CapacityPerToken: Get<Perbill>;
+
+ // ...
```

Comment:
Provider Boosting additions detailed in provider boosting docs, so left out here.